packages/babel-plugin-transform-block-scoping/src/index.ts

```diff
@@ -207,43 +222,60 @@ const letReferenceFunctionVisitor = traverse.visitors.merge([
     },
   },
   tdzVisitor,
-]);
+] as Visitor<LetReferenceVisitorState>[]);
 
-const hoistVarDeclarationsVisitor = {
+const hoistVarDeclarationsVisitor: Visitor<BlockScoping> = {
   enter(path, self) {
-    const { node, parent } = path;
-
     if (path.isForStatement()) {
-      if (isVar(node.init, node)) {
+      const { node } = path;
+      if (isVar(node.init)) {
         const nodes = self.pushDeclar(node.init);
         if (nodes.length === 1) {
           node.init = nodes[0];
         } else {
           node.init = t.sequenceExpression(nodes);
         }
       }
-    } else if (path.isFor()) {
-      if (isVar(node.left, node)) {
+    } else if (path.isForInStatement() || path.isForOfStatement()) {
+      const { node } = path;
+      if (isVar(node.left)) {
         self.pushDeclar(node.left);
         node.left = node.left.declarations[0].id;
       }
-    } else if (isVar(node, parent)) {
+    } else if (isVar(path.node)) {
       path.replaceWithMultiple(
-        self.pushDeclar(node).map(expr => t.expressionStatement(expr)),
+        self.pushDeclar(path.node).map(expr => t.expressionStatement(expr)),
       );
     } else if (path.isFunction()) {
       return path.skip();
     }
   },
 };
 
-const loopLabelVisitor = {
+interface LoopVisitorState {
+  inSwitchCase: boolean;
+  hasBreakContinue: boolean;
+  innerLabels: any[];
+  hasReturn: boolean;
+  ignoreLabeless: boolean;
+  LOOP_IGNORE: symbol;
+  isLoop: boolean;
+  map: any;
+}
+
+const loopLabelVisitor: Visitor<LoopVisitorState> = {
   LabeledStatement({ node }, state) {
     state.innerLabels.push(node.label.name);
   },
 };
 
-const continuationVisitor = {
+interface ContinuationVisitorState {
+  returnStatements: any[];
```

Comment:
Can we improve the types here?
```suggestion
  returnStatements: t.NodePath<t.ReturnStatement>[];
```

Reply:
looks good - added this, and other field types in `ContinuationVisitorState`